### PR TITLE
Decompile NZ0 EntityMariaCutscene

### PR DESCRIPTION
**Pull request by bismurphy**
Wow. This is perhaps one of the hardest functions I've ever done. It does disgusting things with pointers that I've never seen anything like before - some of them are the fault of the programmers, some are the compiler.

Once I was like 98% done, I found func_801B69F8 which is very similar (and decompiled), but wasn't picked up by the duplicate finder, so that was a little sad. Once I found that though, the last of my mysteries were quickly solved.

This is a huge function and it's nice to get it cracked.

### CODE REVIEW

File: src/st/nz0/mariacutscene.c [hunk 1]
```diff
@@ -0,0 +1,728 @@
...
+            D_801CB684 = 1;
+            g_api.FreePrimitives(self->primIndex);
+            self->flags ^= FLAG_HAS_PRIMS;
+            if (g_Dialogue.primIndex[1] != -1) {
+                g_api.FreePrimitives(g_Dialogue.primIndex[1]);
+            }
+            if (g_Dialogue.primIndex[0] != -1) {
+                g_api.FreePrimitives(g_Dialogue.primIndex[0]);
+            }
+            g_api.PlaySfx(0xA);
```
Comment (Xeeynamo) [inline src/st/nz0/mariacutscene.c:278]:
```suggestion
            g_api.PlaySfx(SET_STOP_MUSIC);
```

File: src/st/nz0/mariacutscene.c [hunk 2]
```diff
@@ -0,0 +1,728 @@
...
+            self->step = 1;
+            self->step_s = 0;
+        }
+        if ((self->step) && (g_Dialogue.unk3C != 0)) {
+            func_801B79A8();
+        }
+    }
+    switch (self->step) {
+    case 0:
+        if ((g_CastleFlags[133] != 0) || (g_PlayableCharacter != 0)) {
```
Comment (Xeeynamo) [inline src/st/nz0/mariacutscene.c:288]:
```suggestion
        if ((g_CastleFlags[133] != 0) || (g_PlayableCharacter != PLAYER_ALUCARD)) {
```

File: src/st/sel/3642C.c [hunk 3]
```diff
@@ -236,6 +236,7 @@ void func_801B690C(u8 ySteps, Entity* self) {
     g_Dialogue.portraitAnimTimer++;
 }
 
+// Strong resemblence to NZ0 EntityMariaCutscene and others
```
Comment (Xeeynamo) [inline src/st/sel/3642C.c:239]: _copy&paste intensifies_

**Review by Xeeynamo — APPROVED** [review]
What a goliath of a function. That pointer magic surely is gonna give us problems... For now let's make these small cosmetic changes and put this in the closet before it does anything weird lol

**Comment by bismurphy** [comment]
Thanks, all done!

**Pull request merged**